jenkins/scripts/VersionSelectorScript.groovy

```diff
@@ -0,0 +1,94 @@
+def canBuild(nodeMajorVersion, builderLabel, buildType) {
+
+  // Linux
+  if (builderLabel.indexOf('centos5') == 0 && nodeMajorVersion >= 8)
+    return false
+  if (buildType == 'release' && builderLabel.indexOf('centos6') == 0 && nodeMajorVersion < 8)
+    return false
+  if (builderLabel =~ /centos[67]-(arm)?(64|32)-gcc48/ && nodeMajorVersion >= 10)
+    return false
+  if (builderLabel =~ /centos[67]-(arm)?(64|32)-gcc6/ && nodeMajorVersion < 10)
+    return false
+  if (builderLabel.indexOf('ubuntu1804') == 0 && nodeMajorVersion < 10) // probably temporary
+    return false
+  if (builderLabel.indexOf('ubuntu1204') == 0 && nodeMajorVersion >= 10)
+    return false
```

Comment:
I'm not _sure_ about this, so just a suggestion, but it might be easier to parse if we grouped these by Node version as well as arch, i.e.:

```groovy
  if (builderLabel =~ /centos[67]-(arm)?(64|32)-gcc48/ && nodeMajorVersion >= 10)
    return false
  if (builderLabel =~ /centos[67]-(arm)?(64|32)-gcc6/ && nodeMajorVersion < 10)
    return false
  if (builderLabel.indexOf('ubuntu1804') == 0 && nodeMajorVersion < 10) // probably temporary
    return false
  if (builderLabel.indexOf('ubuntu1204') == 0 && nodeMajorVersion >= 10)
    return false
```

to 

```groovy
if (nodeMajorVersion >= 10) {
  if (builderLabel =~ /centos[67]-(arm)?(64|32)-gcc48/)
    return false
  if (builderLabel =~ /centos[67]-(arm)?(64|32)-gcc6/)
    return false
  if (builderLabel.startsWith('ubuntu1804')) // probably temporary
    return false
  if (builderLabel.startsWith('ubuntu1204'))
    return false
}
```